presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -153,21 +165,25 @@ public HashSymbolPreference translate(Function<Symbol, Optional<Symbol>> transla
             return Optional.of(newSymbols.build());
         }
 
-        public HashSymbolPreference withHashSymbol(PlanNode node, List<Symbol> preferredHashSymbol)
+        public HashSymbols withHashSymbol(PlanNode node, List<Symbol> hashSymbol)
```

Comment:
this means add a symbol for the computation of the hash of the list of source symbols